src/compile.ml

```diff
@@ -62,7 +62,15 @@ let rec compile : bool -> Files.module_path -> unit = fun force path ->
             data.pdata_finalize ss st
       in
       ignore (List.fold_left handle sig_st (parse_file src));
-      if Pervasives.(!gen_obj) then Sign.write sign obj;
+      if Pervasives.(!gen_obj) then
+        begin
+          let not_local _ (sym, _) = Terms.(sym.sym_expo) <> Terms.Local in
+          let sign_symbols =
+            ref (StrMap.filter not_local !(sign.sign_symbols))
+          in
+          let sign = { sign with sign_symbols } in
```

Comment:
Following my comment in https://github.com/Deducteam/lambdapi/pull/262#discussion_r344095185, this should be done in all cases even if gen_obj is not set.